.github/workflows/integration_test.yml

```diff
@@ -0,0 +1,158 @@
+name: Integration Test
+
+on:
+  pull_request:
+    branches: [ "main" ]
+
+jobs:
+  e2e-test:
+    runs-on: macos-13-xl
+    permissions:
+      contents: write
+      checks: write
+      pull-requests: write
+      id-token: write
+    env:
+      iam_role_to_assume: ${{ secrets.ROLE_ARN }}
+      device_farm_project_arn: ${{ secrets.DEVICE_FARM_PROJECT_ARN }}
+      device_farm_pool_arn: ${{ secrets.DEVICE_FARM_POOL_ARN }}
+      device_farm_test_spec_arn: ${{ secrets.DEVICE_FARM_TEST_SPEC_ARN }}
+      clickstream_app_id: ${{ secrets.CLICKSTREAM_APP_ID }}
+      clickstream_endpoint: ${{ secrets.CLICKSTREAM_ENDPOINT }}
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          token: ${{ secrets.PROJECT_TOKEN }}
+      - name: Modify SDK for integration test
+        run: |
+          sed -i '' -e "s#private(set) var bundleSequenceId: Int#private(set) var bundleSequenceId: Int\n    var allEventJson: String = \"\"#g" Sources/Clickstream/Dependency/Clickstream/Analytics/EventRecorder.swift
+          sed -i '' -e "s#toPrettierJsonString())\")#toPrettierJsonString())\")\n            allEventJson.append(\"Saved event \\\(event.eventType):\\\(eventObject.toJsonString())\\\n\")\n            UIPasteboard.general.string = allEventJson#g" Sources/Clickstream/Dependency/Clickstream/Analytics/EventRecorder.swift
+          sed -i '' -e "s#batchEvent.eventCount) events\")#batchEvent.eventCount) events\")\n                allEventJson.append(\"Send \\\(batchEvent.eventCount) events\\\n\")\n                UIPasteboard.general.string = allEventJson#g" Sources/Clickstream/Dependency/Clickstream/Analytics/EventRecorder.swift
+          git diff
+          git config user.name "github-actions"
+          git config user.email "github-actions@github.com"
+          branch=pastboard-$(openssl rand -hex 5)
+          echo $branch
+          echo "temp_branch=$branch" >> "$GITHUB_ENV"
+          git checkout -b $branch
+          git add .
+          git commit -m "ci: integration test"
+          git push --set-upstream origin $branch
+      - name: Prepare sample iOS app
+        run: |
+          git clone https://github.com/aws-samples/clickstream-sdk-samples
+          cd clickstream-sdk-samples/ios
+          sed -i '' -e "s#\"appId\": \"your appId\"#\"appId\": \"${{ env.clickstream_app_id }}\"#g" ModerneShopping/amplifyconfiguration.json
+          sed -i '' -e "s#\"endpoint\": \"your endpoint\"#\"endpoint\": \"${{ env.clickstream_endpoint }}\"#g" ModerneShopping/amplifyconfiguration.json
+          sed -i '' -e "s#if index==0 || index==1 {}#if index==0 || index==1 {\n                                cart.addToCart(addedProduct: product, quantity: 1)\n                            }#g" ModerneShopping/Views/ProductViews/ProductList.swift
+          sed -i '' -e "s#branch = main;#branch = $temp_branch;#g" ModerneShopping.xcodeproj/project.pbxproj
```

Comment:
It's not a good practice to use a temporary **remote** branch for testing. Why not hack the local code used by the sample app?

Reply:
sure, I will have a try, local integration the SDK could be more complex for code modification.

Reply:
done